frontend/src/pages/UserView.js

```diff
@@ -295,100 +317,80 @@ function UserView() {
+								<TableCell align="right">{posIntField("teaching." + i + ".newDevs", "New Devs")}</TableCell>
+								<TableCell align="right">{posIntField("teaching." + i + ".overloads", "Overloads")}</TableCell>
+								<TableCell align="center">{deleteFromListButton("teaching", i)}</TableCell>
 							</TableRow>
-						</TableHead>
-						<TableBody>
-						
-						{/*Start of table contents - Reading JSON in here?*/}
-						{userData && userData?.teaching ? (userData.teaching.map((row, i) => (
-							<TableRow
-							key={i}
-							>
-								<TableCell component="th" scope="row">
-									{row.semester.fullName}
-								</TableCell>
-							<TableCell align="right">{row.newPreps}</TableCell>
-							<TableCell align="right">{row.newDevs}</TableCell>
-							<TableCell align="right">{row.overloads}</TableCell>
-							</TableRow>
-					))) : null}
+						))) : null}
 					</TableBody>
 					</Table>
-					</TableContainer>
-{/*End of example*/}
-					
+				</TableContainer>
+				{addToListButton("teaching")}
 				</Collapse>
 			</Box>
+			
 			<Box sx={mainBoxFormat}>
 				Service Activity
 				<ExpandCollapseButton isOpen={open4} onClick={() => setOpen4(!open4)} />
+					
 				<Collapse in={open4} timeout="auto" unmountOnExit>
-
-				<TableContainer component={Paper}>
-					<Table sx={{ minWidth: 650, fontSize: 30}} size="small" aria-label="a dense table">
-						<TableHead>
-							<TableRow>
-								<TableCell>Semester:</TableCell>
-								<TableCell align="right">Description</TableCell>
-								<TableCell align="right">Level</TableCell>
-							</TableRow>
-						</TableHead>
+					<TableContainer component={Paper}>
+						<Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
 						<TableBody>
-						
-						{/*Start of table contents - Reading JSON in here? yes*/}
-						{userData && userData?.serviceActivity ? (userData.serviceActivity.map((row, i) => (
-							<TableRow
-							key={i}
-							>
-								<TableCell component="th" scope="row">
-									{row.semester.fullName}
-								</TableCell>
-							<TableCell align="right">{row.description}</TableCell>
-							<TableCell align="right">{row.level.level}</TableCell>
+							{userData && userData?.serviceActivity ? (userData.serviceActivity.map((row, i) => (
+							<TableRow>
+								<TableCell align="right">{freeTextField("serviceActivity." + i + ".semester.fullName", "Semester")}</TableCell>
+								<TableCell align="right">{freeTextField("serviceActivity." + i + ".description", "Description")}</TableCell>
+								<TableCell align="right">{dropdownField("serviceActivity." + i + ".level.level", "Level")}</TableCell>
+								<TableCell align="center">{deleteFromListButton("serviceActivity", i)}</TableCell>
 							</TableRow>
-					))) : null}
-					</TableBody>
-					</Table>
+						))) : null}
+						</TableBody>
+						</Table>
 					</TableContainer>
+					{addToListButton("serviceActivity")}
```

Comment:
all of this is basically just adding editable fields to lists, adding remove from list button, and adding the add from list button (and in that order)